src/Core/Services/TypeHelper.cs

```diff
@@ -108,14 +100,51 @@ public static class TypeHelper
             [SqlDbType.SmallInt] = typeof(short),
             [SqlDbType.SmallMoney] = typeof(decimal),
             [SqlDbType.Text] = typeof(string),
-            [SqlDbType.Time] = typeof(TimeSpan),
+            [SqlDbType.Time] = typeof(TimeOnly),
             [SqlDbType.Timestamp] = typeof(byte[]),
             [SqlDbType.TinyInt] = typeof(byte),
             [SqlDbType.UniqueIdentifier] = typeof(Guid),
             [SqlDbType.VarBinary] = typeof(byte[]),
             [SqlDbType.VarChar] = typeof(string)
         };
 
+        /// <summary>
+        /// Given the system type, returns the corresponding primitive type kind.
+        /// </summary>
+        /// <param name="columnSystemType">Type of the column.</param>
+        /// <returns>EdmPrimitiveTypeKind</returns>
+        /// <exception cref="ArgumentException">Throws when the column</exception>
+        public static EdmPrimitiveTypeKind GetEdmPrimitiveTypeFromSystemType(Type columnSystemType)
+        {
+            if (columnSystemType.IsArray)
+            {
+                columnSystemType = columnSystemType.GetElementType()!;
+            }
+
+            EdmPrimitiveTypeKind type = columnSystemType.Name switch
+            {
+                "String" => EdmPrimitiveTypeKind.String,
+                "Guid" => EdmPrimitiveTypeKind.Guid,
+                "Byte" => EdmPrimitiveTypeKind.Byte,
+                "Int16" => EdmPrimitiveTypeKind.Int16,
+                "Int32" => EdmPrimitiveTypeKind.Int32,
+                "Int64" => EdmPrimitiveTypeKind.Int64,
+                "Single" => EdmPrimitiveTypeKind.Single,
+                "Double" => EdmPrimitiveTypeKind.Double,
+                "Decimal" => EdmPrimitiveTypeKind.Decimal,
+                "Boolean" => EdmPrimitiveTypeKind.Boolean,
+                "DateTime" => EdmPrimitiveTypeKind.DateTimeOffset,
+                "DateTimeOffset" => EdmPrimitiveTypeKind.DateTimeOffset,
+                "Date" => EdmPrimitiveTypeKind.Date,
+                "TimeOnly" => EdmPrimitiveTypeKind.TimeOfDay,
+                "TimeSpan" => EdmPrimitiveTypeKind.TimeOfDay,
```

Comment:
Removed TimeOnly, because when fetching Schema, it is recognizing the Time column as TimeSpan, usually TimeSpan should be mapped with EdmType.Duration, but here we using TimeOfDay because it's for the Time and not Duration.

Reply:
I added tests for OData as well.

Reply:
> Removed TimeOnly, because when fetching Schema, it is recognizing the Time column as TimeSpan,

When would we resolve a SQL column of tsql type `time` to .NET `TimeOnly` as indicated is possible by https://erikej.github.io/dotnet/sqlclient/2022/11/17/sqlclient-dateonly.html

Since we map SqlDbType.Time and DbType.Time to .NET `TimeOnly` we may want to keep `"TimeOnly" => EdmPrimitiveTypeKind.TimeOfDay,` in case we update to support reading db type time to .net TimeOnly type. or if there is a code path today that does that , that we didn't catch here.

Reply:
yep, makes sense.